### PR TITLE
Introduce notion of experimental target, made targets which needs sim…

### PR DESCRIPTION
**Pull request by olonho**
…ulator and Zephyr experimental.

### CODE REVIEW

File: runtime/build.gradle [hunk 1]
```diff
@@ -20,8 +20,6 @@ targetList.each { targetName ->
         dependsOn "${targetName}Debug"
         dependsOn "${targetName}Release"
         target targetName
-	    if (!isWindows())
-	       compilerArgs '-fPIC'
```
Comment (msink) [inline runtime/build.gradle]: So `-fPIC` is not needed for all targets, not just mingw?
It also present in 4 more files:
```
backend.native/build.gradle
backend.native/tests/build.gradle
libclangext/build.gradle
llvmDebugInfoC/build.gradle
```

Comment (olonho) [reply runtime/build.gradle]: Guess it just not needed here.

File: shared/src/main/kotlin/org/jetbrains/kotlin/konan/target/KonanTarget.kt [hunk 2]
```diff
@@ -188,23 +192,37 @@ open class HostManager(protected val distribution: Distribution = Distribution()
...
-                ) + zephyrSubtargets
+                )
                 KonanTarget.MACOS_X64 -> listOf(
                     KonanTarget.MACOS_X64,
                     KonanTarget.IOS_ARM32,
                     KonanTarget.IOS_ARM64,
                     KonanTarget.IOS_X64,
                     KonanTarget.LINUX_X64,
                     KonanTarget.LINUX_ARM32_HFP,
+                    KonanTarget.MINGW_X64,
```
Comment (msink) [inline shared/src/main/kotlin/org/jetbrains/kotlin/konan/target/KonanTarget.kt]: Both here and in experimental?

File: shared/src/main/kotlin/org/jetbrains/kotlin/konan/target/KonanTarget.kt [hunk 3]
```diff
@@ -188,23 +192,37 @@ open class HostManager(protected val distribution: Distribution = Distribution()
...
                     KonanTarget.WASM32
-                ) + zephyrSubtargets
-                else ->
-                    throw TargetSupportException("Unknown host platform: $host")
+                )
+                else -> throw TargetSupportException("Unknown host platform: $host")
             }        
     }
 
+    val enabledExperimental: List<KonanTarget> by lazy {
```
Comment (msink) [inline shared/src/main/kotlin/org/jetbrains/kotlin/konan/target/KonanTarget.kt]: It isn't really enabled - zephyr and mingw on linux did not built by `./gradlew clean build bundle`

Comment (olonho) [reply shared/src/main/kotlin/org/jetbrains/kotlin/konan/target/KonanTarget.kt]: Sure, it's WIP PR.

File: shared/src/main/kotlin/org/jetbrains/kotlin/konan/target/KonanTarget.kt (hunk 2 above)
Comment (olonho) [reply shared/src/main/kotlin/org/jetbrains/kotlin/konan/target/KonanTarget.kt]: Just for development purposes.

**Comment by olonho** [comment]
Obsolete in favour of https://github.com/JetBrains/kotlin-native/pull/2681.

**Pull request closed without merging**